feature(scale 2D): add scale on 2D views

mgermerie opened the pull request:
## Description
<!--- Describe your changes in detail -->
Add a scale div in view_2d_map example.
Add an event triggering in PlanarControls when a view animation is over.
Detect this event in view_2d_map example to compute scale.

mgermerie committed:
feature(scale 2D): trigger event when 2D map animation is over to detect when scale computation is relevant

feature(scale 2D): trigger scale changed event when vertical movement is done (pan) ; refactor the event triggering when the view is zoomed

feature(scale 2D): trigger movement event for all movements

mgermerie committed:
example(map 2D and 25D): add scale on 2D and 2.5D map examples

gchoqueux commented:
Could you add scale widget on 25d example?
`Travel` isn't the only move to change the range.
In this example, you can try other moves.

mgermerie commented:
I added a scale on the 2.5D map example.
Further more, a `RANGE_CHANGED` event is now also triggered when user performs a `pan` movement.

gchoqueux commented on examples/view_25d_map.html:
```diff
@@ -97,6 +99,33 @@
 
+            // Add scale :
+            let divScaleWidget = document.getElementById('divScaleWidget');
+            function updateScaleWidget() {
```
Could you factorize `updateScaleWidget` in `examples/js ` to share the function

gchoqueux commented on src/Controls/PlanarControls.js:
```diff
@@ -219,6 +223,9 @@ function PlanarControls(view, options = {}) {
+            });
             this.view.notifyChange(this.camera);
         }
         if (this.state === STATE.DRAG) {
```
I think for all `STATE` the range will change

mgermerie committed:
examples(scale): factorize the updateScaleWidget method as twoo methods declared in Scale.js

fix(Scale.js): disable eslint unused-var inspection for scale methos

examples(scale): update some docs

examples(scale): modify a parameter name

gchoqueux commented on examples/js/Scale.js:
```diff
@@ -0,0 +1,40 @@
+/**
+ * For a given pixel distance at the center of a given view, calculates the equivalent metric distance.
```
there's not pixel distance from center, it's projected pixels on ground.

gchoqueux commented on examples/view_25d_map.html:
```diff
@@ -97,6 +100,25 @@
             new itowns.PlanarControls(view, {});
 
+            // Initialize scale :
+            let initialScaleSize = 200;
```
it's constant

gchoqueux reviewed (commented):
Could squash all fix commits?

It's feature, not a fix, no?

gchoqueux commented on examples/js/Scale.js:
```diff
@@ -0,0 +1,40 @@
+ * @return {{pixels: number, meters: number}} The rounded pixel and metric distances.
+ */
+// eslint-disable-next-line no-unused-vars
+function roundPixelsFromMeters(view, distancePixels) {
```
it's not distancePixels

gchoqueux commented on examples/view_25d_map.html:
```diff
@@ -97,6 +100,25 @@
             new itowns.PlanarControls(view, {});
 
+            // Initialize scale :
+            const initialScaleSize = 200;
```
could you add `in pixel`

gchoqueux commented:
could you rebase on master?
